src/Controls/src/Core/Handlers/Items/CarouselViewHandler.Windows.cs

```diff
@@ -548,10 +549,31 @@ void Resize(Size newSize)
 			{
 				_currentSize = newSize;
 
-				UpdateItemsSource();
-				UpdateSnapPointsType();
-				UpdateSnapPointsAlignment();
-				UpdateCarouselViewInitialPosition();
+				if (_isCarouselViewReady)
+					InvalidateItemSize();
+				else
+					InitialSetup();
+
+				_isCarouselViewReady = true;
+			}
+		}
+
+		void InitialSetup()
+		{
+			UpdateItemsSource();
+			UpdateSnapPointsType();
+			UpdateSnapPointsAlignment();
+			UpdateCarouselViewInitialPosition();
+		}
+
+		void InvalidateItemSize()
+		{
+			foreach (var item in ListViewBase.GetChildren<ItemContentControl>())
+			{
+				item.ItemHeight = GetItemHeight();
+				item.ItemWidth = GetItemWidth();
```

Comment:
It's beyond the scope of this PR, but I wonder why this logic isn't in the layout/measure pass. You'd think that item width + height is just calculated based on horizontal/vertical modes